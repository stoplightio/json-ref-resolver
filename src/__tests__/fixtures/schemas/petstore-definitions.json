{
  "input": {
    "Pet": {
      "type": "object",
      "required": [
        "name",
        "photoUrls"
      ],
      "properties": {
        "id": {
          "type": "integer",
          "format": "int64"
        },
        "category": {
          "$ref": "#/definitions/Category"
        },
        "name": {
          "type": "string",
          "example": "doggie"
        },
        "photoUrls": {
          "type": "array",
          "xml": {
            "wrapped": true
          },
          "items": {
            "type": "string",
            "xml": {
              "name": "photoUrl"
            }
          }
        },
        "tags": {
          "type": "array",
          "xml": {
            "wrapped": true
          },
          "items": {
            "xml": {
              "name": "tag"
            },
            "$ref": "#/definitions/Tag"
          }
        },
        "status": {
          "type": "string",
          "description": "pet status in the store",
          "enum": [
            "available",
            "pending",
            "sold"
          ]
        }
      },
      "xml": {
        "name": "Pet"
      }
    },
    "Order": {
    "type": "object",
    "xml": {
      "name": "Order"
    },
    "properties": {
      "id": {
        "type": "integer",
        "format": "int64"
      },
      "petId": {
        "type": "integer",
        "format": "int64"
      },
      "quantity": {
        "$ref": "./todos-circular-definitions.json#/definitions/todo-partial"
      },
      "shipDate": {
        "type": "string",
        "format": "date-time"
      },
      "status": {
        "type": "string",
        "description": "Order Status",
        "enum": [
          "placed",
          "approved",
          "delivered"
        ]
      },
      "complete": {
        "type": "boolean"
      }
    }
  },
    "User": {
    "type": "object",
    "properties": {
      "id": {
        "type": "integer",
        "format": "int64"
      },
      "username": {
        "type": "string"
      },
      "firstName": {
        "type": "string"
      },
      "lastName": {
        "type": "string"
      },
      "email": {
        "type": "string"
      },
      "password": {
        "type": "string"
      },
      "phone": {
        "type": "string"
      },
      "userStatus": {
        "type": "integer",
        "format": "int32",
        "description": "User Status"
      }
    },
    "xml": {
      "name": "User"
    }
  }
  },
  "expected": {
    "Order": {
      "properties": {
        "complete": {
          "type": "boolean"
        },
        "id": {
          "format": "int64",
          "type": "integer"
        },
        "petId": {
          "format": "int64",
          "type": "integer"
        },
        "quantity": {
          "properties": {
            "completed": {
              "type": [
                "boolean",
                "null"
              ]
            },
            "name": {
              "$ref": "#/definitions/todo-full"
            }
          },
          "required": [
            "name",
            "completed"
          ],
          "title": "Todo Partial",
          "type": "object"
        },
        "shipDate": {
          "format": "date-time",
          "type": "string"
        },
        "status": {
          "description": "Order Status",
          "enum": [
            "placed",
            "approved",
            "delivered"
          ],
          "type": "string"
        }
      },
      "type": "object",
      "xml": {
        "name": "Order"
      }
    },
    "Pet": {
      "properties": {
        "category": {
          "$ref": "#/definitions/Category"
        },
        "id": {
          "format": "int64",
          "type": "integer"
        },
        "name": {
          "example": "doggie",
          "type": "string"
        },
        "photoUrls": {
          "items": {
            "type": "string",
            "xml": {
              "name": "photoUrl"
            }
          },
          "type": "array",
          "xml": {
            "wrapped": true
          }
        },
        "status": {
          "description": "pet status in the store",
          "enum": [
            "available",
            "pending",
            "sold"
          ],
          "type": "string"
        },
        "tags": {
          "items": {
            "$ref": "#/definitions/Tag",
            "xml": {
              "name": "tag"
            }
          },
          "type": "array",
          "xml": {
            "wrapped": true
          }
        }
      },
      "required": [
        "name",
        "photoUrls"
      ],
      "type": "object",
      "xml": {
        "name": "Pet"
      }
    },
    "User": {
      "properties": {
        "email": {
          "type": "string"
        },
        "firstName": {
          "type": "string"
        },
        "id": {
          "format": "int64",
          "type": "integer"
        },
        "lastName": {
          "type": "string"
        },
        "password": {
          "type": "string"
        },
        "phone": {
          "type": "string"
        },
        "userStatus": {
          "description": "User Status",
          "format": "int32",
          "type": "integer"
        },
        "username": {
          "type": "string"
        }
      },
      "type": "object",
      "xml": {
        "name": "User"
      }
    }
  }
}
