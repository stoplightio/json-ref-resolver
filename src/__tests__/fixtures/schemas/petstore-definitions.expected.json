{
  "ApiResponse": {
    "properties": {
      "code": {
        "format": "int32",
        "type": "integer"
      },
      "message": {
        "type": "string"
      },
      "type": {
        "type": "string"
      }
    },
    "type": "object"
  },
  "Category": {
    "properties": {
      "id": {
        "format": "int64",
        "type": "integer"
      },
      "name": {
        "type": "string"
      }
    },
    "type": "object",
    "xml": {
      "name": "Category"
    }
  },
  "Order": {
    "properties": {
      "complete": {
        "type": "boolean"
      },
      "id": {
        "format": "int64",
        "type": "integer"
      },
      "petId": {
        "format": "int64",
        "type": "integer"
      },
      "quantity": {
        "properties": {
          "completed": {
            "type": [
              "boolean",
              "null"
            ]
          },
          "name": {
            "$ref": "#/definitions/todo-full"
          }
        },
        "required": [
          "name",
          "completed"
        ],
        "title": "Todo Partial",
        "type": "object"
      },
      "shipDate": {
        "format": "date-time",
        "type": "string"
      },
      "status": {
        "description": "Order Status",
        "enum": [
          "placed",
          "approved",
          "delivered"
        ],
        "type": "string"
      }
    },
    "type": "object",
    "xml": {
      "name": "Order"
    }
  },
  "Pet": {
    "properties": {
      "category": {
        "$ref": "#/definitions/Category"
      },
      "id": {
        "format": "int64",
        "type": "integer"
      },
      "name": {
        "example": "doggie",
        "type": "string"
      },
      "photoUrls": {
        "items": {
          "type": "string",
          "xml": {
            "name": "photoUrl"
          }
        },
        "type": "array",
        "xml": {
          "wrapped": true
        }
      },
      "status": {
        "description": "pet status in the store",
        "enum": [
          "available",
          "pending",
          "sold"
        ],
        "type": "string"
      },
      "tags": {
        "items": {
          "$ref": "#/definitions/Tag",
          "xml": {
            "name": "tag"
          }
        },
        "type": "array",
        "xml": {
          "wrapped": true
        }
      }
    },
    "required": [
      "name",
      "photoUrls"
    ],
    "type": "object",
    "xml": {
      "name": "Pet"
    }
  },
  "Tag": {
    "properties": {
      "id": {
        "$ref": "#/definitions/Order"
      },
      "name": {
        "type": "string"
      }
    },
    "type": "object",
    "xml": {
      "name": "Tag"
    }
  },
  "User": {
    "properties": {
      "email": {
        "type": "string"
      },
      "firstName": {
        "type": "string"
      },
      "id": {
        "format": "int64",
        "type": "integer"
      },
      "lastName": {
        "type": "string"
      },
      "password": {
        "type": "string"
      },
      "phone": {
        "type": "string"
      },
      "userStatus": {
        "description": "User Status",
        "format": "int32",
        "type": "integer"
      },
      "username": {
        "type": "string"
      }
    },
    "type": "object",
    "xml": {
      "name": "User"
    }
  }
}
